appinventor/components/src/com/google/appinventor/components/runtime/Form.java

```diff
@@ -695,8 +706,29 @@ void ReplayFormOrientation() {
     }
   }
 
+  private Integer generateHashCode(AndroidViewComponent component, PercentStorageRecord.Dim dim) {
+    if (dim == PercentStorageRecord.Dim.HEIGHT) {
+      return (component.hashCode() + dim.hashCode()) * 2 + 1;
```

Comment:
You don't need to add `dim.hashCode()` here (or the next line). By virtue of adding 0 or 1 we are encoding the dimension information perfectly.